pkg/providers/nutanix/template.go

```diff
@@ -519,3 +524,26 @@ func generateNutanixFailureDomains(eksNutanixFailureDomains []v1alpha1.NutanixDa
 	}
 	return failureDomains
 }
+
+func addKubeVipToIgnoredNodeIPsList(clusterSpec *cluster.Spec, result []string) []string {
+	kubeVipStr := clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host
+	if kubeVipStr != "" {
+		kubeVip, err := net.ResolveIPAddr("ip", kubeVipStr)
+		if err != nil {
+			// log error and continue
+			log.Printf("error resolving kube-vip IP address %s: %v", kubeVipStr, err)
+		} else {
+			result = append(result, kubeVip.IP.String())
+		}
+	}
+
+	return result
+}
+
+func generateCcmIgnoredNodeIPsList(clusterSpec *cluster.Spec) []string {
+	result := make([]string, 0)
+
+	result = addKubeVipToIgnoredNodeIPsList(clusterSpec, result)
+
+	return result
```

Comment:
With the above suggestion, I would recommend getting rid of `addKubeVipToIgnoredNodeIPsList` function and just do something like this instead
```suggestion
        ignoredIPs := []string{clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host}
	return ignoredIPs
```

Reply:
Fixed